shared/login/register/code-page/qr/index.ios.js

```diff
@@ -2,27 +2,55 @@
 import Camera from 'react-native-camera'
 import React, {Component} from 'react'
 import type {Props} from './index'
-import {NativeImage, Box} from '../../../../common-adapters/index.native'
+import {NativeImage, Box, Text} from '../../../../common-adapters/index.native'
 import {globalStyles} from '../../../../styles'
 import {throttle} from 'lodash'
 
-class QR extends Component<void, Props, void> {
+class QR extends Component<void, Props, {permissionGranted: ?boolean}> {
+  state = {
+    permissionGranted: null,
+  }
+
+  componentDidMount() {
+    this._requestCameraPermission()
+  }
+
+  async _requestCameraPermission() {
+    try {
+      const permissionGranted = await Camera.checkVideoAuthorizationStatus()
+      this.setState({permissionGranted})
+    } catch (err) {
+      console.warn(err)
```

Comment:
Might be worth doing `console.warn('Error checking camera auth:', err)` so it's easier to find if this happens.